### PR TITLE
feat: Implement DataStore for Storing Application Preferences

### PR DESCRIPTION
**Pull request by aslamhossin**
- Implemented DataStore for storing application preferences in the common, iOS, and Android modules.
 - Provided DataStore dependency injection using Koin for easy access across the application.
 - Implemented a single application using androidx.startup for streamlined initialization.
 - Defined a simple use case in PaginatedHomeViewModel for demonstrating ViewModel usage with DataStore.

### CODE REVIEW

File: composeApp/src/androidMain/kotlin/data/preference/Preference.android.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+/**
+ * Platform-specific function for creating a DataStore instance.
+ * Creates a DataStore instance for Android platform.
+ *
+ * @return DataStore<Preferences> instance.
+ */
+
+internal actual fun createDataStore(): DataStore<Preferences> = createDataStore(
+    producePath = { applicationContext.filesDir.resolve(DATA_STORE_FILE_NAME).absolutePath }
+)
```
Comment (ShaQib07) [inline composeApp/src/androidMain/kotlin/data/preference/Preference.android.kt]: Missing new line

File: composeApp/src/commonMain/kotlin/data/preference/Preference.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+fun createDataStore(producePath: () -> String): DataStore<Preferences> {
+    return PreferenceDataStoreFactory.createWithPath(
+        corruptionHandler = null,
+        migrations = emptyList(),
+        scope = CoroutineScope(Dispatchers.Default + SupervisorJob()), // Coroutine scope for IO operations
+        produceFile = { producePath().toPath() }, // Produces the file path for the DataStore
+    )
+}
+
+internal expect fun createDataStore(): DataStore<Preferences>
```
Comment (ShaQib07) [inline composeApp/src/commonMain/kotlin/data/preference/Preference.kt]: missing new line

File: composeApp/src/commonMain/kotlin/data/preference/SingleDataStorePreferences.kt [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            }
+        } catch (e: SerializationException) {
+            print("$e")
+        }
+    }
+
+    override suspend fun clear() {
+        dataStore.edit { it[key] = "" }
+    }
+}
```
Comment (ShaQib07) [inline composeApp/src/commonMain/kotlin/data/preference/SingleDataStorePreferences.kt]: missing new line

File: composeApp/src/commonMain/kotlin/domain/usecase/GetUserFromPreferenceUseCase.kt [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import data.preference.SingleDataStorePreferences
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class User(val name: String, val email: String)
+class GetUserFromPreferenceUseCase(dataStore: DataStore<Preferences>) :
+    SingleDataStorePreferences<User>(dataStore, User.serializer())
```
Comment (ShaQib07) [inline composeApp/src/commonMain/kotlin/domain/usecase/GetUserFromPreferenceUseCase.kt]: missing new line

File: composeApp/src/iosMain/kotlin/data/preference/Preference.ios.kt [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+            inDomain = NSUserDomainMask,
+            appropriateForURL = null,
+            create = false,
+            error = null
+        )
+
+        // Constructs the full file path for the DataStore
+        requireNotNull(docDirectory).path + "/$DATA_STORE_FILE_NAME"
+    }
+)
```
Comment (ShaQib07) [inline composeApp/src/iosMain/kotlin/data/preference/Preference.ios.kt]: missing new line

**Review by ShaQib07 — COMMENTED** [review]
Left some comments

**Review by ShaQib07 — APPROVED** [review]
LGTM

**Review by jhasan95 — APPROVED** [review]
LGTM!
As we discussed, 
- Add unit test in the next PR on this.
- Store login tokens and establish a token-based user session in the next PR.

**Pull request merged**